### PR TITLE
Bluetooth: Controller: Fix filter accept list and privacy default, and DUP_FILTER_ADV_SET_MAX dependency

### PR DESCRIPTION
**Pull request by cvinayak**
Fix filter accept list and privacy feature Kconfig default
based on whether host has them enabled.

Fix missing BT_CTLR_DUP_FILTER_ADV_SET_MAX dependency on
BT_CTLR_ADV_EXT.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/openisa/lll/lll_adv.c [hunk 1]
```diff
@@ -608,20 +608,21 @@ static void isr_rx(void *param)
...
 		devmatch_ok = radio_filter_has_match();
 		devmatch_id = radio_filter_match_get();
-		irkmatch_ok = radio_ar_has_match();
-		irkmatch_id = radio_ar_match_get();
+		if (IS_ENABLED(CONFIG_BT_CTLR_PRIVACY)) {
+			irkmatch_ok = radio_ar_has_match();
+			irkmatch_id = radio_ar_match_get();
+		} else {
+			irkmatch_ok = 0U;
+			irkmatch_id = 0xFF;
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/openisa/lll/lll_adv.c:616]: Do we have a `#define` for this value? I assume it's some kind of placeholder to denote that there isn't a match?

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -157,7 +157,7 @@ config BT_CTLR_DUP_FILTER_LEN
 
 config BT_CTLR_DUP_FILTER_ADV_SET_MAX
 	int "Number of Extended Advertising Sets in the scan duplicate filter"
-	depends on BT_OBSERVER && (BT_CTLR_DUP_FILTER_LEN > 0)
+	depends on BT_OBSERVER && BT_CTLR_ADV_EXT && (BT_CTLR_DUP_FILTER_LEN > 0)
```
Comment (Thalley) [inline subsys/bluetooth/controller/Kconfig:160]: Why does this need to depend on `BT_CTLR_ADV_EXT`? Are we not using the same duplicate filter list for legacy scanning?

Comment (cvinayak) [reply subsys/bluetooth/controller/Kconfig:160]: This value is used only when Extended Scanning is supported. Additional values like DID, data status etc. is maintained to filter duplicate Extended Advertising PDUs from same peer device that uses multiple advertising sets.

File: subsys/bluetooth/controller/ll_sw/openisa/lll/lll_adv.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/openisa/lll/lll_adv.c]: Fixed

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**